src/components/common/ColonyActions/helpers/getEventTitleValues.ts

```diff
@@ -49,11 +51,22 @@ enum EventTitleMessageKeys {
   ChainName = 'chainName',
   SafeAddress = 'safeAddress',
   RemovedSafes = 'removedSafes',
+  SafeName = 'safeName',
+  SafeTransactionAmount = 'safeTransactionAmount',
+  IsSafeTransactionRecipientUser = 'isSafeTransactionRecipientUser',
+  SafeTransactionRecipient = 'safeTransactionRecipient',
+  SafeTransactionAddress = 'safeTransactionAddress',
+  SafeTransactionContractName = 'safeTransactionContractName',
+  SafeTransactionFunctionName = 'safeTransactionFunctionName',
+  SafeTransactionNftToken = 'safeTransactionNftToken',
```

Comment:
so it doesnt seem to be enough overlap to enable reusing the ones we already have for instance reusing `Recipient`.

The `Recipient` getter in `mapColonyActionToExpectedFormat` is defined as:
```ts
    recipient: (
      <span className={styles.titleDecoration}>
        {actionData.recipientUser ? (
          <FriendlyName user={actionData.recipientUser} autoShrinkAddress />
        ) : (
          <MaskedAddress address={actionData.recipientAddress || AddressZero} />
        )}
      </span>
    ),
```
now we could use the try to use the recipient in the first safe transaction ie 
```ts
        {actionData.recipientUser ||
        actionData.safeTransaction?.transactions[0].recipient ? (
```
however that data field is very sparsely filled out at the moment:
```json
[
    {
        "__typename": "SafeTransactionData",
        "transactionType": "TRANSFER_NFT",
        "token": null,
        "amount": null,
        "rawAmount": "undefined",
        "recipient": {
            "__typename": "SimpleTarget",
            "id": "0xb77D57F4959eAfA0339424b83FcFaf9c15407461",
            "profile": {
                "__typename": "SimpleTargetProfile",
                "avatarHash": null,
                "displayName": null,
                "username": null
            },
            "walletAddress": "0xb77D57F4959eAfA0339424b83FcFaf9c15407461"
        },
        "data": "",
        "contract": null,
        "abi": "",
        "contractFunction": "safeTransferFrom",
        "nft": {
            "__typename": "NFT",
            "id": "0xB4feDc003053C22ac8b808Bb424f3e1787f30cF2 16009",
            "profile": {
                "__typename": "NFTProfile",
                "displayName": "JungleVIBES #16009"
            },
            "walletAddress": "0xB4feDc003053C22ac8b808Bb424f3e1787f30cF2"
        },
        "nftData": {
            "__typename": "NFTData",
            "address": "0xB4feDc003053C22ac8b808Bb424f3e1787f30cF2",
            "description": "Created by NFTSTAR \n\nAuthorized by Neymar Jr.",
            "id": "16009",
            "imageUri": "https://cdn.nftstar.com/neymar/junglevibes/images/jungleVibes.jpeg",
            "logoUri": "https://safe-transaction-assets.safe.global/tokens/logos/0xB4feDc003053C22ac8b808Bb424f3e1787f30cF2.png",
            "name": "JungleVIBES #16009",
            "tokenName": "NRJJungleVIBES",
            "tokenSymbol": "NRJJV",
            "uri": "https://cdn.nftstar.com/neymar/junglevibes/metadata/16009"
        },
        "functionParams": []
    }
]
```
 so we need to fetch the data from the context using the `useUserByAddress` hook which requires a separate entry. Is it possible to save the user profiles in the saga? Then we could remove this duplication

Reply:
I now see that the `isSafeTransactionRecipientUser` and switch between username and fallback to address is a pattern that has been changed in the move to the CDapp from the Dapp. So I have removed 
```ts
  IsSafeTransactionRecipientUser = 'isSafeTransactionRecipientUser',
  SafeTransactionRecipient = 'safeTransactionRecipient',
  SafeTransactionAddress = 'safeTransactionAddress',
```
and use the recipient key to format this correctly, however we still are missing the recipient user data in the db

Reply:
Well, it would depend on the recipient. The recipients used by transfer funds, transfer nft and raw transaction should be in the DB if the user chooses a user from the picker. In the case of the contract interaction, yeah the recipient is a contract so we shouldn't have it in the db.